pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/BuildWheelsTask.groovy

```diff
@@ -114,17 +124,37 @@ class BuildWheelsTask extends DefaultTask {
      */
     private void buildWheels(Project project, Collection<File> installables, PythonDetails pythonDetails) {
 
+        ProgressLoggerFactory progressLoggerFactory = getServices().get(ProgressLoggerFactory)
+        ProgressLogger progressLogger = progressLoggerFactory.newOperation(BuildWheelsTask)
+        progressLogger.setDescription("Building Wheels")
+        progressLogger.started()
+
         WheelExtension wheelExtension = ExtensionUtils.getPythonComponentExtension(project, WheelExtension)
         def pythonExtension = ExtensionUtils.getPythonExtension(project)
 
+        PythonTag pythonTag = PythonTag.findTag(getProject(), getPythonDetails())
+        PlatformTag platformTag = PlatformTag.makePlatformTag(getProject(), getPythonDetails())
+
+        WheelCache wheelCache = new WheelCache(wheelCacheDir, pythonTag, platformTag)
+
+        def taskTimer = new TaskTimer()
+
+        int counter = 0
         installables.each { File installable ->
 
             def packageInfo = PackageInfo.fromPath(installable.path)
             def shortHand = packageInfo.version ? "${packageInfo.name}-${packageInfo.version}" : packageInfo.name
-            def messageHead = 'Preparing wheel ' + shortHand
+
+            def clock = taskTimer.start(shortHand)
+            progressLogger.progress("Preparing wheel $shortHand (${++counter} of ${installables.size()})")
+
+            def wheel = wheelCache.findWheel(packageInfo.name, packageInfo.version, pythonExtension.details.getPythonVersion())
+            if (wheel.isPresent()) {
+                FileUtils.copyFile(wheel.get(), new File(wheelExtension.wheelCache, wheel.get().name))
+                return
+            }
 
             if (packageExcludeFilter.isSatisfiedBy(packageInfo)) {
-                logger.lifecycle(PythonHelpers.createPrettyLine(messageHead, "[EXCLUDED]"))
```

Comment:
Do we still need logging on some level of these packages that have been excluded so that we can debug/reason about them and whether the exclusion was correct?

Reply:
Added ```if (pythonExtension.consoleOutput == ConsoleOutput.RAW) {
                LOGGER.lifecycle("Installing {}", shortHand)
            }
``` that should happen on CI machines.